drivers/include/net/netdev2.h

```diff
@@ -10,15 +10,153 @@
+ * through reading and setting device parameters, and (iii) event handling.
+ *
+ *
+ * @section sec_problem The Interrupt Context Problem
+ *
+ * Network devices are typically connected to the host CPU via some sort of
+ * electrical bus, most commonly via SPI. This type of connection has the
+ * disadvantage, that the bus is not used by the network device alone, but it
+ * may be shared with other devices. This makes it necessary to synchronize
+ * access to the bus to prevent bus access collisions.
+ *
+ * To illustrate this behavior, let's look at a typical error situation, that
+ * leads to a very hard to find and debug latent failure: say we have two
+ * devices A and B on the same SPI bus. Our CPU is now transferring a chunk of
+ * 100 bytes to device A. After 20 bytes were transferred, device B triggers
+ * an external interrupt on the host CPU. The interrupt handling now typically
+ * requires the reading of some sort of status register on the 'triggering'
+ * device, device B in this case. So what would happen here, is that the device
+ * driver for device B would initiate a new SPI transfer on the already used bus
+ * to read B's status register -> BAM.
+ *
+ * The peripheral drivers for shared buses (i.e. SPI and I2C) implement access
+ * synchronization using mutexes, which are locked and unlocked in the driver's
+ * `require` and `release` functions. The problem is now, that this type of
+ * synchronization does only work in thread context, but not in interrupt
+ * context. With reasonable effort and resource usage, we have no means of
+ * synchronizing the bus access also in interrupt context.
+ *
+ * The solution to this problem as implemented by this interface is **not to
+ * call any function that interacts with a device directly from interrupt
+ * context**. Unfortunately this requires some added complexity for
+ * synchronization efforts between thread and interrupt context to be able to
+ * handle device events (i.e. external interrupts). See section @res sec_events
+ * for more information.
+ *
+ *
+ * @section sec_context Context requirements
+ *
+ * The `netdev` interface expects the network device drivers to run in thread
+ * context (see section above). The interface was however designed in a ways, to
+ * allow more than one device driver to be serviced in the same thread.
+ *
+ * The key design element for `netdev` is, that device drivers implementing this
+ * interface are not able to run stand-alone in a thread, but need some
+ * bootstrapping code. This bootstrapping code can be anything from a simple
+ * msg_receive() loop (as done for the GNRC adaption) to a complete network
+ * stack that works without messaging entirely but is build on function call
+ * interfaces.
+ *
+ *
+ * @section sec_sendrecv Sending and Receiving
+ *
+ * Sending data using the `netdev` interface is straight forward: simply call
+ * the drivers send() function, passing it the data that should be send. The
```

Comment:
For all occurences: `@ref netdev2_driver_t::send "send()"` (otherwise `send()` in `sys/posix/include/sys/socket.h` gets referenced). Same goes for `receive()`.